### PR TITLE
add image builder based on disko

### PR DESCRIPTION
**Pull request by Lassulus**
like make-disk-image.nix from nixpkgs, but cooler :)
(multi image support, takes disko config, etc.)

### CODE REVIEW

**Comment by Lassulus** [comment]
it can be used like this: https://github.com/Lassulus/flakes-testing/blob/master/flake.nix#L11

File: lib/default.nix [hunk 1]
```diff
@@ -10,6 +10,9 @@ let
 
   diskoLib = {
 
+    # like make-disk-image.nix from nixpkgs, but with disko config
+    makeDiskImage = args: import ./make-disk-image.nix ({ inherit diskoLib; } // args);
```
Comment (Mic92) [inline lib/default.nix:14]: Can we also add a test or example in our flake?

File: lib/make-disk-image.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+pkgs.vmTools.runInLinuxVM (pkgs.runCommand name {
+  buildInputs = with pkgs; [
+    systemdMinimal
+    nixos-install-tools
+    nix
+    utillinux
+  ];
+  preVM = ''
+    # TODO: get size either dynamically or from disko config
+    ${lib.concatMapStringsSep "\n" (disk: "truncate -s 15G ${disk.name}.raw") (lib.attrValues nixosConfig.config.disko.devices.disk)}
```
Comment (Mic92) [inline lib/make-disk-image.nix:25]: Should we make this configurable at least?

File: lib/make-disk-image.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  QEMU_OPTS = lib.concatMapStringsSep " " (disk: "-drive file=${disk.name}.raw,if=virtio,cache=unsafe,werror=report") (lib.attrValues nixosConfig.config.disko.devices.disk);
+
+  memSize = 1024;
+} ''
+  # running udev, stolen from stage-1.sh
+  echo "running udev..."
+  ln -sfn /proc/self/fd /dev/fd
+  ln -sfn /proc/self/fd/0 /dev/stdin
+  ln -sfn /proc/self/fd/1 /dev/stdout
+  ln -sfn /proc/self/fd/2 /dev/stderr
```
Comment (Mic92) [inline lib/make-disk-image.nix:40]: Do we have other common devices such as /dev/null?

File: lib/make-disk-image.nix [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    }];
+  };
+in
+pkgs.vmTools.runInLinuxVM (pkgs.runCommand name {
+  buildInputs = with pkgs; [
+    systemdMinimal
+    nixos-install-tools
+    nix
+    utillinux
+  ];
```
Comment (Mic92) [inline lib/make-disk-image.nix:22]: When people have common hooks, we probably should expose an extraPackages parameter?

Comment (Mic92) [reply lib/make-disk-image.nix:22]: Maybe not needed.

File: lib/make-disk-image.nix (hunk 3 above)
Comment (Lassulus) [reply lib/make-disk-image.nix:42]: dunno, this is taken from nixos stage-1

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**